dist/schema-form.js

```diff
@@ -226,6 +231,67 @@ angular.module('schemaForm').provider('schemaFormDecorators',
               }
             };
 
+            var evalExpression = function (expression) {
+              angular.forEach(scope.form.dependencies, function (key) {
+                expression = expression.replace(key, lookupForKey(key));
+              });
+
+              return scope.$eval(expression);
+            };
+
+
+            var resetModelValue = function () {
+              if (scope.ngModel) {
+                scope.ngModel.$setViewValue(undefined);
+                scope.ngModel.$render();
+                scope.ngModel.$setPristine();
+              }
+            };
+
+            scope.showCondition = function () {
+              var expressionString = scope.form.expression;
+              if (angular.isUndefined(expressionString)) {
+                return true;
+              }
+
+              var show = evalExpression(expressionString);
+
+              if (angular.isDefined(scope.form.required)) {
+                scope.form.required = show;
+                scope.form.schema.required = show;
+              }
+
+
+              if (scope.form.key && !show) {
+                resetModelValue();
+              }
+
+              return show;
+
+            };
+
+
+            scope.disabledElement = function () {
+              var expressionString = scope.form.disabledExpression;
+
+              if (angular.isUndefined(expressionString)) {
+                return false;
+              }
+
+              var disabled = evalExpression(expressionString);
+
+              if (angular.isDefined(scope.form.required)) {
+                scope.form.required = !disabled;
+                scope.form.schema.required = !disabled;
```

Comment:
Yes, I fill awkward for this part. But maybe you can give me an advice, how to handle better situation when field is required only if it is visible. Don't you?